### PR TITLE
add clickhouse dialect

### PR DESCRIPTION
**Pull request by gusinacio**
This PR add a way to abstract parts of the database connection called `dialects` inspired by the solution from [jimsmart/schema](https://github.com[/jimsmart/schema](https://github.com/jimsmart/schema)).

The first two dialects that are being added are:
- postgres (the current default)
- clickhouse

The goal is to keep adding methods to dialects depending on each database needs. Please let me know what you think about this solution. Feel free to request changes in the code organization and maybe some better solution on how to keep default values.

### CODE REVIEW

File: db/operations.go [hunk 1]
```diff
@@ -51,6 +53,7 @@ func (l *Loader) newUpdateOperation(table *TableInfo, primaryKey map[string]stri
 		opType:     OperationTypeUpdate,
 		primaryKey: primaryKey,
 		data:       data,
+		dialect:    l.getDialect(),
```
Comment (maoueh) [inline db/operations.go]: I think I prefer that we don't put dialect on the operation directly. I think `func (o *Operation) query()` should receive the dialect.

File: db/operations.go [hunk 2]
```diff
@@ -132,7 +136,7 @@ func getPrimaryKeyWhereClause(primaryKey map[string]string) string {
 	return strings.Join(reg[:], " AND ")
 }
 
-func prepareColValues(table *TableInfo, colValues map[string]string) (columns []string, values []string, err error) {
+func prepareColValues(table *TableInfo, colValues map[string]string, d dialect) (columns []string, values []string, err error) {
```
Comment (maoueh) [inline db/operations.go]: Let's make the dialect the first parameter in this file.

File: docker-compose.yml [hunk 3]
```diff
@@ -33,3 +33,15 @@ services:
       - DATABASE_URL=postgres://dev-node:insecure-change-me-in-prod@postgres:5432/dev-node?sslmode=disable
     depends_on:
       - postgres
+  database:
```
Comment (maoueh) [inline docker-compose.yml:36]: We might want to start having multiple different files, and `up.sh` would take the correct identifier. We can do it later also.

**Review by maoueh — CHANGES_REQUESTED** [review]
Seems pretty solid, not sure if you are planning more PR, will be missing changelog udpate before final merge to `develop`

File: go.mod [hunk 4]
```diff
@@ -154,3 +163,5 @@ require (
 	google.golang.org/grpc v1.54.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1
 )
+
+replace github.com/jimsmart/schema => github.com/semiotic-ai/schema v0.0.0-20230727171059-24e630184512
```
Comment (maoueh) [inline go.mod:167]: Could you show me in the sink code PR above where this fork is in application.

File: db/operations.go (hunk 1 above)
Comment (gusinacio) [reply db/operations.go]: Thank you for your suggestion, I wasn't liking this solution either.

File: db/operations.go (hunk 2 above)
Comment (gusinacio) [reply db/operations.go]: Done

File: go.mod (hunk 4 above)
Comment (gusinacio) [reply go.mod:167]: db.go, lines: 103, 139, 185

Comment (gusinacio) [reply go.mod:167]: Basically, the `Tables` and `PrimaryKeys` functions trigger a query that look for specific tables depending on which database driver is being used. It already has mysql, mssql, postgres, etc I just added clickhouse

**Comment by gusinacio** [comment]
I'm planning some follow-on PRs, because some methods aren't supported by clickhouse and/or have different statement (like update). Also, there are some capabilities missing for clickhouse like `inject-csv` command which I find better to review in another PR.

**Comment by maoueh** [comment]
All good!

**Pull request merged**